scipy/fft/_helper.py

```diff
@@ -140,40 +143,6 @@ def prev_fast_len(target, real=False):
 prev_fast_len.__signature__ = _sig_prev_fast_len
 
 
-def _init_nd_shape_and_axes(x, shape, axes):
-    """Handle shape and axes arguments for N-D transforms.
-
-    Returns the shape and axes in a standard form, taking into account negative
-    values and checking for various potential errors.
-
-    Parameters
-    ----------
-    x : array_like
-        The input array.
-    shape : int or array_like of ints or None
-        The shape of the result. If both `shape` and `axes` (see below) are
-        None, `shape` is ``x.shape``; if `shape` is None but `axes` is
-        not None, then `shape` is ``numpy.take(x.shape, axes, axis=0)``.
-        If `shape` is -1, the size of the corresponding dimension of `x` is
-        used.
-    axes : int or array_like of ints or None
-        Axes along which the calculation is computed.
-        The default is over all axes.
-        Negative indices are automatically converted to their positive
-        counterparts.
-
-    Returns
-    -------
-    shape : tuple
-        The shape of the result as a tuple of integers.
-    axes : list
-        Axes along which the calculation is computed, as a list of integers.
-
-    """
-    x = np.asarray(x)
-    return _helper._init_nd_shape_and_axes(x, shape, axes)
```

Comment:
does pocketfft handle array-likes internally? What motivated this change?

Reply:
The motivation is line 173 above: np.asarray(x) fails on cupy or other arrays on non CPU devices.

The pocketfft helper only looks at .shape and ndim attributes AFAICS.

A separate question is why we keep carrying the pocketfft C++ code instead of calling into numpy (these days: for numpy arrays only). But this is definitely separate.

Reply:
okay, if we always call it with arrays then that's fine - maybe would be nice to try to contribute the docstring Peter wrote upstream rather than just deleting it?

> A separate question is why we keep carrying the pocketfft C++ code instead of calling into numpy (these days: for numpy arrays only). But this is definitely separate.

yeah, I think we can go ahead with that once our minimum NumPy is 2.0

Reply:
If there's anything to upstream, it'd be the helper which accepts ndarrays only to use their `.ndim` and `.shape` atrributes. Whether this is worth the churn upstream, I personally doubt.

Reply:
ah, it looks like the `helper.py` module is not vendored, rather original to SciPy. So could you just move the docstring to the implementation instead?

Reply:
OK, done.